server/metrics/requests.go

```diff
@@ -22,82 +22,51 @@ import (
-	ServerRequestsHandledTotal       = "server_requests_handled_total"
-	ServerRequestsUnknownErrorTotal  = "server_requests_unknown_error_total"
-	ServerRequestsSpecificErrorTotal = "server_requests_specific_error_total"
-	ServerRequestsOkTotal            = "server_requests_ok_total"
-	ServerRequestsHandlingTimeBucket = "server_requests_handling_time_bucket"
-)
-
 var (
-	InitializedServerRequestsCounterNames = []string{
-		ServerRequestsReceivedTotal,
-		ServerRequestsHandledTotal,
-		ServerRequestsUnknownErrorTotal,
-		ServerRequestsOkTotal,
-	}
-	ServerRequestsHistogramNames = []string{
-		ServerRequestsHandlingTimeBucket,
-	}
+	preInitCounterNames              = []string{"received", "handled", "ok", "unknown_error"}
+	preInitSpecificErrorSources      = []string{"tigris_server", "fdb_server"}
+	RequestResponseTimeHistogramName = "handling_time_bucket"
 )
 
-type ServerEndpointMetadata struct {
-	grpcServiceName string
-	grpcMethodName  string
-	grpcTypeName    string
+type RequestEndpointMetadata struct {
+	serviceName string
+	methodInfo  grpc.MethodInfo
 }
 
-type ServerRequestCounter struct {
-	Name    string
-	Tags    map[string]string
-	Counter tally.Counter
+func newRequestEndpointMetadata(serviceName string, methodInfo grpc.MethodInfo) RequestEndpointMetadata {
+	return RequestEndpointMetadata{serviceName: serviceName, methodInfo: methodInfo}
 }
 
-type ServerRequestHistogram struct {
-	Name      string
-	Tags      map[string]string
-	Histogram tally.Histogram
-}
-
-func newServerEndpointMetadata(grpcServiceName string, grpcMethodName string, grpcTypeName string) ServerEndpointMetadata {
-	return ServerEndpointMetadata{grpcServiceName: grpcServiceName, grpcMethodName: grpcMethodName, grpcTypeName: grpcTypeName}
+func getRequestTypeFromMethodInfo(methodInfo grpc.MethodInfo) string {
+	if methodInfo.IsServerStream {
+		return "stream"
+	} else {
+		return "unary"
+	}
 }
 
-func (g *ServerEndpointMetadata) getTags() map[string]string {
+func (g *RequestEndpointMetadata) GetPreInitializedTags() map[string]string {
 	return map[string]string{
-		"method":  g.grpcMethodName,
-		"service": g.grpcServiceName,
-		"type":    g.grpcTypeName,
+		"tigris_server_request_method":       g.methodInfo.Name,
+		"tigris_server_request_service_name": g.serviceName,
+		"tigris_server_request_type":         getRequestTypeFromMethodInfo(g.methodInfo),
 	}
 }
 
-func (g *ServerEndpointMetadata) getSpecificErrorTags(source string, code string) map[string]string {
+func (g *RequestEndpointMetadata) GetSpecificErrorTags(source string) map[string]string {
 	return map[string]string{
-		"method":  g.grpcMethodName,
-		"service": g.grpcServiceName,
-		"type":    g.grpcTypeName,
-		"source":  source,
-		"code":    code,
+		"tigris_server_request_method":       g.methodInfo.Name,
+		"tigris_server_request_service_name": g.serviceName,
+		"tigris_server_request_type":         getRequestTypeFromMethodInfo(g.methodInfo),
+		"tigris_server_request_error_source": source,
```

Comment:
do you need source here? can we have source as a subscope on root scope? i.e. `tigris.server` or `tigris.fdb`?